### PR TITLE
Added thumb mode support under IAR for module manager on Cortex-A7 pl…

### PR DESCRIPTION
**Pull request by yajunxiaMS**
…atform.

## PR checklist

- [ x] Updated function header with a short description and version number
- [ x] Added test case for bug fix or new feature
- [ x] Validated on real hardware

### CODE REVIEW

File: ports_module/cortex_a7/iar/example_build/sample_threadx_module.c [hunk 1]
```diff
@@ -232,11 +232,12 @@ void    thread_0_entry(ULONG thread_input)
 
 UINT    status;
 
+#if 0  /* TBD: Disabled external memory test as the MMU need to be refactored. */
```
Comment (TiejunMS) [inline ports_module/cortex_a7/iar/example_build/sample_threadx_module.c:235]: Is this change intended?

File: ports_module/cortex_a7/gnu/module_manager/src/tx_thread_stack_build.s [hunk 2]
```diff
@@ -38,6 +38,8 @@ CPSR_MASK       =       0xFF                    // Mask initial CPSR, T, IRQ & F
 CPSR_MASK       =       0xBF                    // Mask initial CPSR, T, IRQ interrupts enabled
 #endif
 
+    .global     _tx_thread_schedule
```
Comment (TiejunMS) [inline ports_module/cortex_a7/gnu/module_manager/src/tx_thread_stack_build.s:41]: Should `.global` be replaced with `EXTERN` here?

File: ports_module/cortex_a7/iar/example_build/sample_threadx_module.c (hunk 1 above)
Comment (yajunxiaMS) [reply ports_module/cortex_a7/iar/example_build/sample_threadx_module.c]: removed, resolved.

File: ports_module/cortex_a7/iar/example_build/sample_threadx_module.ewp [hunk 3]
```diff
@@ -509,7 +509,7 @@
                 </option>
                 <option>
                     <name>ADefines</name>
-                    <state></state>
+                    <state>THUMB_MODE</state>
```
[the remark this answers is not in the document]
Comment (yajunxiaMS) [reply ports_module/cortex_a7/iar/example_build/sample_threadx_module.ewp]: accept, resloved.

File: ports_module/cortex_a7/iar/example_build/txm.ewp [hunk 4]
```diff
@@ -509,7 +509,7 @@
                 </option>
                 <option>
                     <name>ADefines</name>
-                    <state></state>
+                    <state>THUMB_MODE</state>
```
[the remark this answers is not in the document]
Comment (yajunxiaMS) [reply ports_module/cortex_a7/iar/example_build/txm.ewp]: resoved

File: ports_module/cortex_a7/gnu/module_manager/src/tx_thread_stack_build.s (hunk 2 above)
Comment (yajunxiaMS) [reply ports_module/cortex_a7/gnu/module_manager/src/tx_thread_stack_build.s:41]: [ports_module/cortex_a7/gnu/module_manager/src/tx_thread_context_save.s] has been fixed

File: ports_module/cortex_a7/iar/example_build/sample_threadx_module_manager.ewp [hunk 5]
```diff
@@ -509,7 +509,7 @@
                 </option>
                 <option>
                     <name>ADefines</name>
-                    <state></state>
+                    <state>THUMB_MODE</state>
```
[the remark this answers is not in the document]
Comment (yajunxiaMS) [reply ports_module/cortex_a7/iar/example_build/sample_threadx_module_manager.ewp]: accept, resloved.

File: ports_module/cortex_a7/iar/example_build/tx.ewp [hunk 6]
```diff
@@ -515,6 +515,7 @@
                 <option>
                     <name>ADefines</name>
                     <state>noTX_ENABLE_EXECUTION_CHANGE_NOTIFY</state>
+                    <state>THUMB_MODE</state>
```
[the remark this answers is not in the document]
Comment (yajunxiaMS) [reply ports_module/cortex_a7/iar/example_build/tx.ewp]: accept, resloved.

**Review by TiejunMS — APPROVED** [review]

**Review by xiuwencai — APPROVED** [review]
